lib/utils/getFormatter.mjs

```diff
@@ -0,0 +1,64 @@
+import { EXIT_CODE_INVALID_CONFIG } from '../constants.mjs';
+import dynamicImport from './dynamicImport.mjs';
+import { existsSync } from 'node:fs';
+import formatters from '../formatters/index.mjs';
+import getConfigForFile from '../getConfigForFile.mjs';
+import getFormatterOptionsText from './getFormatterOptionsText.mjs';
+import { resolve } from 'node:path';
+
+/** @import {Formatter, FormatterType, InternalApi} from 'stylelint' */
+
+/**
+ * @param {InternalApi} stylelint
+ * @param {Object} [params]
+ * @param {FormatterType | Formatter | undefined} [params.formatterType]
+ * @returns {Promise<Formatter>}
+ */
+export default async function getFormatter(stylelint) {
+	const cwd = stylelint._options.cwd;
+	const configPath = stylelint._options.configFile || cwd;
+	let formatter = stylelint._options.formatter;
+
+	if (!formatter) {
+		let configForFile;
+
+		try {
+			configForFile = await getConfigForFile(stylelint, configPath);
+		} catch (err) {
+			if (err instanceof Error && 'code' in err && err.code === EXIT_CODE_INVALID_CONFIG) {
+				configForFile = undefined;
+			} else {
+				throw err;
+			}
+		}
+
+		const resolvedConfig = stylelint._options.config ? stylelint._options : configForFile;
+
+		formatter = resolvedConfig?.config.formatter;
```

Comment:
[suggestion] `stylelint._options.config` seems no longer needed. Can we simplify the code like this?

```suggestion
		formatter = configForFile?.config.formatter;
```